source/split-words.d.ts

```diff
@@ -36,22 +39,30 @@ export type SplitWords<
 		? [...SkipEmptyWord<CurrentWord>, ...SplitWords<RemainingCharacters>]
 		: LastCharacter extends ''
 			// Fist char of word
-			? SplitWords<RemainingCharacters, FirstCharacter, FirstCharacter>
-			// Case change: non-numeric to numeric, push word
+			? SplitWords<RemainingCharacters, Options, FirstCharacter, FirstCharacter>
+			// Case change: non-numeric to numeric
 			: [false, true] extends [IsNumeric<LastCharacter>, IsNumeric<FirstCharacter>]
-				? [...SkipEmptyWord<CurrentWord>, ...SplitWords<RemainingCharacters, FirstCharacter, FirstCharacter>]
-				// Case change: numeric to non-numeric, push word
+				// Split on number: push word
+				? Options['splitOnNumber'] extends true
```

Comment:
When there's a numeric/non-numeric case change I added and used a new option `splitOnNumber` which depending on how it is set will either
1. When set to true - Like before split words on number
2. When set to false - concat to current word